### PR TITLE
Charlie shift crud create page

### PR DESCRIPTION
**Pull request by cjyu81**
Create page for CRUD Driver Shift Table.

### CODE REVIEW

**Comment by cjyu81** [comment]
I will push a fix for this. Dont review yet

**Comment by cjyu81** [comment]
Fixes pushed. Ready for review

**Review by adityapatil834 — DISMISSED** [review]

**Review by pconrad — APPROVED** [review]
LGTM

**Review by chriscerie — CHANGES_REQUESTED** [review]

File: frontend/src/main/pages/Shift/ShiftCreatePage.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  const { isSuccess } = mutation
+
+  const onSubmit = async (data) => {
+    mutation.mutate(data);
+  }
+
+  if (isSuccess && !storybook) {
+    return <Navigate to="/shift" />
+  }
```
[the remark this answers is not in the document]
Comment (cjyu81) [reply frontend/src/main/pages/Shift/ShiftCreatePage.js:43]: Moving it into the event handler causes the following error:
image

The example code RestaurantCreatePage.js from last lab uses this format so I'm not sure how to proceed with your suggested changes.

[the remark this answers is not in the document]
Comment (chriscerie) [reply frontend/src/main/pages/Shift/ShiftCreatePage.js:43]: > The example code RestaurantCreatePage.js from last lab uses this format

I hope to see this change then. Logic driven primarily from user input should be in event handlers separate from effects and render logic.

As for your code, it doesn't make sense to return elements from event handlers. Navigate imperatively with `useNavigate`.

[the remark this answers is not in the document]
Comment (pconrad) [reply frontend/src/main/pages/Shift/ShiftCreatePage.js:43]: @chriscerie Raises some good points, but those probably need to be incorporated into our stack and our examples at an earlier stage in the course.  For now, we'll let this go in as is, and I'll take @chriscerie 's points under consideration for refactoring the examples for F23—and if that goes well, we might create issues to refactor the legacy apps along the same lines.

File: frontend/src/tests/pages/Shift/ShiftCreatePage.test.js [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        fireEvent.click(createButton);
+
+        // Wait for the axios call to be made
+        await waitFor(() => expect(axiosMock.history.post.length).toBe(1));
+
+        // Asserting that the post request was made with correct parameters
+        expect(axiosMock.history.post[0].params).toEqual(noidshift);
+
+        // Assert that the toast and navigate functions were called with expected values
+        //expect(mockToast).toBeCalledWith(`New shift Created - id: ${shift.id} day: ${shift.day}`);
```
Comment (pconrad) [inline frontend/src/tests/pages/Shift/ShiftCreatePage.test.js:126]: Oof.. commented out code.  This should be removed before we merge.

**Review by pconrad — CHANGES_REQUESTED** [review]
Found a line of commented out code that should be removed.

File: frontend/src/tests/pages/Shift/ShiftCreatePage.test.js (hunk 2 above)
Comment (pconrad) [inline frontend/src/tests/pages/Shift/ShiftCreatePage.test.js:126]:
```suggestion
```

**Comment by cjyu81** [comment]
Comment deleted. Ready for review

**Review by pconrad — APPROVED** [review]
LGTM!

**Pull request merged**